Initial commit for CI

lucernae opened the pull request:
- Integrate Travis
- Integrate codecov

lucernae committed:
Initial commit for CI

- Integrate Travis
- Integrate codecov

lucernae commented:
Weird, travis seems to be building a queue of draft PR. So now, the build is trying to catch up.

Anyway, what needs to be done:

- [ ] Fix flake error. It made travis unhappy
- [x] Secret.py does not automatically generated. It's blocking everything.

pierrealixt commented on .travis.yml:
```diff
@@ -0,0 +1,53 @@
+        - echo "Docker stack ready"
+        - echo "Preparing unittests"
+        - make flake8
+        - echo "Run unittests"
```
`make test` is already present

lucernae replied:
Yes, I saw that you create another Make command, so I didn't make a new one here to avoid conflict.

lucernae committed:
Add Django Secret Generation for Travis CI

Module core.settings.secret.py doesn't exists by default, but is needed
by Django. We will attempt to generate this module first before
'migrate' and 'collectstatic' command when using this in Travis.

We also replace test command in travis with `make test`

lucernae committed:
Fix STATIC_URL not set for production

A misconfiguration of prod settings makes the stack unable to
complete collectstatic

lucernae committed:
Fix flake8 error for settings file

lucernae committed:
Disable cache build for Travis

Context change is working already, but we need to rethink how
are we going to store docker image specifically built to run
Travis test

lucernae commented:
I'm done with this PR @pierrealixt @Christiaanvdm . We should Review + Merge this ASAP to avoid further conflicts with current develop. Improvements should be done in separate PR. This PR contains basic mechanism of running Travis CI config.

It will still be red, since most of the test doesn't pass yet. But we should make separate PR to fix each test.

pierrealixt reviewed (approved):
LVGTM
